akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -110,7 +110,7 @@ private[akka] object Running {
       case PoisonPill =>
         if (isInternalStashEmpty && !isUnstashAllInProgress) Behaviors.stopped
         else new HandlingCommands(state.copy(receivedPoisonPill = true))
-      case signal =>
+      case signal if setup.isSignalDefined(state.state, signal) =>
```

Comment:
good, then there are several more places that have the same problem. look for calls to `setup.onSignal`